Azure Micronaut Data MySQL Guide

graemerocher opened the pull request:
To build the guide

```bash
./gradlew micronautCloudDatabaseAzureBuild
open build/dist/index.html 
```

burtbeckwith committed:
allow apps to have no features

burtbeckwith committed:
micronaut-cloud-database base guide and Oracle Cloud guide

burtbeckwith committed:
Merge branch 'master' into micronaut-cloud-database

burtbeckwith committed:
remove slug and asciidoctor

burtbeckwith committed:
add ingress rule step and curl examples

burtbeckwith committed:
only check output dir exists once

burtbeckwith committed:
updated apps merge logic

retain apps that are only in the base metadata or only in guide metadata, then merge features from apps in both

burtbeckwith committed:
Merge branch 'master' into micronaut-cloud-database

burtbeckwith committed:
delete .gitkeep

sdelamo committed:
Update metadata.json (#768)

burtbeckwith committed:
PR feedback from Hari

highlight that there's no free-tier MySQL
add PowerShell cli info

graemerocher committed:
Add azure guide

graemerocher committed:
correct output

burtbeckwith commented on guides/micronaut-cloud-database-azure/micronaut-cloud-database-azure.adoc:
```diff
@@ -0,0 +1,197 @@
+
+== Create an Azure Database for MySQL instance
+
+The first step in this guide is to create a new https://azure.microsoft.com/en-us/services/mysql/[Azure Database for MySQL] instance. 3 steps are requires to achieve that using the https://docs.microsoft.com/en-us/cli/azure/install-azure-cli[Azure CLI]:
```
requires -> required

graemerocher committed:
typo

burtbeckwith commented on guides/micronaut-cloud-database-azure/micronaut-cloud-database-azure.adoc:
```diff
@@ -0,0 +1,197 @@
+external:micronaut-cloud-database-base/start.adoc[]
+
+We will then connect the application to https://azure.microsoft.com/en-us/services/mysql/[Azure Database for MySQL].
```
this should probably go after the account note so it doesn't split up the reqs list, or we could rework start.adoc

burtbeckwith commented on guides/micronaut-cloud-database-azure/micronaut-cloud-database-azure.adoc:
```diff
@@ -0,0 +1,197 @@
+
+== Create an Azure Database for MySQL instance
+
+The first step in this guide is to create a new https://azure.microsoft.com/en-us/services/mysql/[Azure Database for MySQL] instance. 3 steps are required to achieve that using the https://docs.microsoft.com/en-us/cli/azure/install-azure-cli[Azure CLI]:
```
3 -> three (generally spell out numbers below 10)

burtbeckwith commented on guides/micronaut-cloud-database-azure/micronaut-cloud-database-azure.adoc:
```diff
@@ -0,0 +1,197 @@
+
+The first step in this guide is to create a new https://azure.microsoft.com/en-us/services/mysql/[Azure Database for MySQL] instance. 3 steps are required to achieve that using the https://docs.microsoft.com/en-us/cli/azure/install-azure-cli[Azure CLI]:
+
+1. First we need to create a resource group that the database will form part of. It is recommended to create a new resource group just for this guide and not re-use an existing one.
```
I'd remove "need to" in each of these, e.g. "First, create a ...", "Then, create the Azure ..."

burtbeckwith committed:
remove Oracle Cloud from shared

burtbeckwith committed:
Proofreading fixes

burtbeckwith committed:
update publication date

graemerocher committed:
Merge branch 'micronaut-cloud-database' into azure-database-guide

graemerocher committed:
Merge branch 'master' into azure-database-guide

graemerocher committed:
address feedback

graemerocher commented:
@sdelamo this one is ready for review / proof reading etc.

shingjo commented on guides/micronaut-cloud-database-azure/micronaut-cloud-database-azure.adoc:
```diff
@@ -0,0 +1,196 @@
+
+[source, bash]
+----
+curl http://localhost:/genres/list
```
Missing port number.
`curl http://localhost:8080/genres/list`

shingjo commented on guides/micronaut-cloud-database-azure/metadata.json:
```diff
@@ -0,0 +1,8 @@
+  "base": "micronaut-cloud-database-base",
+  "title": "Connect a Micronaut Data JDBC application to Azure Database for MySQL",
+  "intro": "Learn how to connect to Azure Database for MySQL",
+  "tags": ["Azure"],
```
Add Author here instead of the base metadata:
"authors": ["Graeme Rocher"],

shingjo commented on guides/micronaut-cloud-database-azure/metadata.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "base": "micronaut-cloud-database-base",
+  "title": "Connect a Micronaut Data JDBC application to Azure Database for MySQL",
```
The phrase "Azure Database *for* MySQL" sounds strange to me. 
What about "Connect a Micronaut Data JDBC application to an Azure MySQL Database"

graemerocher replied:
it may sound weird but it is what Azure calls it https://azure.microsoft.com/en-us/services/mysql/#overview

sdelamo committed:
Merge branch 'main' into azure-database-guide

sdelamo committed:
add github workflow guides

sdelamo commented:
@kroegerk can you proofread it?

sdelamo committed:
remove Graeme Rocher as base author

sdelamo committed:
add localhost:8080

sdelamo committed:
Merge branch 'master' into azure-database-guide

kroegerk commented on guides/micronaut-cloud-database-azure/metadata.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "base": "micronaut-cloud-database-base",
+  "authors": ["Graeme Rocher"],
+  "title": "Connect a Micronaut Data JDBC application to Azure Database for MySQL",
```
```suggestion
  "title": "Connect a Micronaut Data JDBC Application to Azure Database for MySQL",
```
title case

kroegerk commented on guides/micronaut-cloud-database-azure/metadata.json:
```diff
@@ -0,0 +1,9 @@
+  "base": "micronaut-cloud-database-base",
+  "authors": ["Graeme Rocher"],
+  "title": "Connect a Micronaut Data JDBC application to Azure Database for MySQL",
+  "intro": "Learn how to connect to Azure Database for MySQL",
```
```suggestion
  "intro": "Learn how to connect a Micronaut Data JDBC application to a Microsoft Azure Database for MySQL",
```

kroegerk reviewed (commented):
Please see my revised version of the document, submitted as a pull request.

graemerocher committed:
Apply suggestions from code review

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

kroegerk committed:
Proofreading (#797)

The instructions seemed to be out of order. It said the first thing the reader would do was to create a resource group, but then it jumped backwards and said the first thing to do was configure environment variables. I reordered the sections so the environment variables come first, to make it more clear.

Other changes are general corrections and recommendations for clarity and consistency.

sdelamo committed:
revert sections

sdelamo committed:
change azureregion copy
